x-pack/packages/kbn-ecs-data-quality-dashboard/impl/data_quality/data_quality_panel/body/data_quality_details/storage_details/helpers.ts

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Datum, Key, ArrayNode } from '@elastic/charts';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { sortBy } from 'lodash/fp';
+
+import { getSizeInBytes } from '../../../../helpers';
+import { getIlmPhase } from '../../../pattern/helpers';
+import { PatternRollup } from '../../../../types';
+
+export interface LegendItem {
+  color: string | null;
+  ilmPhase: string | null;
+  index: string | null;
+  pattern: string;
+  sizeInBytes: number;
+}
+
+export interface FlattenedBucket {
+  ilmPhase: string | undefined;
+  incompatible: number | undefined;
+  indexName: string | undefined;
+  pattern: string;
+  sizeInBytes: number;
+}
+
+export const getPatternSizeInBytes = ({
+  pattern,
+  patternRollups,
+}: {
+  pattern: string;
+  patternRollups: Record<string, PatternRollup>;
+}): number => {
+  if (patternRollups[pattern] != null) {
+    return patternRollups[pattern].sizeInBytes ?? 0;
+  } else {
+    return 0;
+  }
+};
+
+export const getPatternLegendItem = ({
+  pattern,
+  patternRollups,
+}: {
+  pattern: string;
+  patternRollups: Record<string, PatternRollup>;
+}): LegendItem => ({
+  color: null,
+  ilmPhase: null,
+  index: null,
+  pattern,
+  sizeInBytes: getPatternSizeInBytes({ pattern, patternRollups }),
+});
+
+export const getLegendItemsForPattern = ({
+  pattern,
+  flattenedBuckets,
+}: {
+  pattern: string;
+  flattenedBuckets: FlattenedBucket[];
+}): LegendItem[] =>
+  sortBy(
```

Comment:
You can use `orderBy` to sort in descending order without the additional `reverse`. Not sure which is more performant though